airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadStreamStateManager.java

```diff
@@ -0,0 +1,113 @@
+    this.pairToPrimaryKeyInfo = pairToPrimaryKeyInfo;
+    this.pairToPrimaryKeyLoadStatus = initPairToPrimaryKeyLoadStatusMap(initialLoadStreams.pairToInitialLoadStatus());
+    this.streamsThatHaveCompletedSnapshot = initStreamsCompletedSnapshot(initialLoadStreams, catalog);
+  }
+
+  @Override
+  public void updatePrimaryKeyLoadState(final io.airbyte.protocol.models.AirbyteStreamNameNamespacePair pair,
+                                        final PrimaryKeyLoadStatus pkLoadStatus) {
+    pairToPrimaryKeyLoadStatus.put(pair, pkLoadStatus);
+  }
+
+  public AirbyteStateMessage createFinalStateMessage(final io.airbyte.protocol.models.AirbyteStreamNameNamespacePair pair,
+                                                     final JsonNode streamStateForIncrementalRun) {
+    streamsThatHaveCompletedSnapshot.add(pair);
+
+    return new AirbyteStateMessage()
+        .withType(AirbyteStateType.STREAM)
+        .withData(Jsons.jsonNode(getFinalState(pair)))
+        .withStream(getAirbyteStreamState(pair, (streamStateForIncrementalRun)));
+  }
+
+  @Override
+  public PrimaryKeyInfo getPrimaryKeyInfo(final io.airbyte.protocol.models.AirbyteStreamNameNamespacePair pair) {
+    return pairToPrimaryKeyInfo.get(pair);
+  }
+
+  @Override
+  public PrimaryKeyLoadStatus getPrimaryKeyLoadStatus(final io.airbyte.protocol.models.AirbyteStreamNameNamespacePair pair) {
+    return pairToPrimaryKeyLoadStatus.get(pair);
+  }
+
+  @Override
+  public AirbyteStateMessage createIntermediateStateMessage(final io.airbyte.protocol.models.AirbyteStreamNameNamespacePair pair,
+                                                            final PrimaryKeyLoadStatus pkLoadStatus) {
+    return new AirbyteStateMessage()
+        .withType(AirbyteStateType.STREAM)
+        .withStream(getAirbyteStreamState(pair, Jsons.jsonNode(pkLoadStatus)));
+  }
+
+
+
+  private AirbyteStreamState getAirbyteStreamState(final io.airbyte.protocol.models.AirbyteStreamNameNamespacePair pair, final JsonNode stateData) {
+    LOGGER.info("STATE DATA FOR {}: {}", pair.getNamespace().concat("_").concat(pair.getName()), stateData);
+    assert Objects.nonNull(pair);
+    assert Objects.nonNull(pair.getName());
+    assert Objects.nonNull(pair.getNamespace());
+
+    return new AirbyteStreamState()
+        .withStreamDescriptor(
+            new StreamDescriptor().withName(pair.getName()).withNamespace(pair.getNamespace()))
+        .withStreamState(stateData);
+  }
+
+
+
+  private DbStreamState getFinalState(final io.airbyte.protocol.models.AirbyteStreamNameNamespacePair pair) {
+    assert Objects.nonNull(pair);
+    assert Objects.nonNull(pair.getName());
+    assert Objects.nonNull(pair.getNamespace());
+
+    return new DbStreamState()
+        .withStreamName(pair.getName())
+        .withStreamNamespace(pair.getNamespace())
+        .withCursorField(Collections.emptyList())
```

Comment:
why does the final state has `cursorFiled` empty and `cursor` `null`? This seems wrong!

Reply:
I dont think this is required, we can get rid of this